File: backend/src/main/java/com/inboxreader/backend/service/BookmarkService.java
package com.inboxreader.backend.service;

import com.inboxreader.backend.dto.response.BookmarkResponse;
import java.util.List;

public interface BookmarkService {
    BookmarkResponse toggleBookmark(Long userId, Long articleId);
    List<Long> getBookmarkedArticleIds(Long userId);
}

File: frontend/src/services/bookmarkService.ts
import api from "./api";
import type { BookmarkResponse } from "../types/bookmark";

export const toggleBookmark = async (
  userId: number,
  articleId: number,
): Promise<BookmarkResponse> => {
  const response = await api.post<BookmarkResponse>("/bookmarks/toggle", null, {
    params: { userId, articleId },
  });
  return response.data;
};

export const getBookmarkedArticleIds = async (
  userId: number = 1,
): Promise<number[]> => {
  const response = await api.get<number[]>("/bookmarks", {
    params: { userId },
  });
  return response.data;
};
